app/plugins/shards/shards.js

```diff
@@ -0,0 +1,238 @@
+        '<div class="cr-custom-tooltip-label"> REST URL </div>' +
+        '<div> <i>' + rest_url + '</i> </div>' +
+        '</div>';
+    }
+
+    function getHtmlStringForShard(id, size, min_lucene_version, num_docs, primary, relocating_node, relocating_state, state, shard_class) {
+      return '<div class="row"> ' +
+        '<div class="shard-status ' + shard_class + '"></div>' +
+        '<div class="cr-custom-tooltip-section"> ' +
+        '<div class="cr-custom-tooltip-label"> ID </div>' +
+        '<div> <i>' + id + '</i> </div>' +
+        '<div class="cr-custom-tooltip-label"> Size </div>' +
+        '<div> <i>' + size + '</i> </div>' +
+        '<div class="cr-custom-tooltip-label"> Min Lucene Version </div>' +
+        '<div> <i>' + min_lucene_version + '</i> </div>' +
+        '<div class="cr-custom-tooltip-label"> Num docs </div>' +
+        '<div> <i>' + num_docs + '</i> </div>' +
+        '</div> ' +
+        '<div class="cr-custom-tooltip-section "> ' +
+        '<div class="cr-custom-tooltip-label"> Primary Shard </div>' +
+        '<div> <i>' + primary + '</i> </div>' +
+        '<div class="cr-custom-tooltip-label"> Relocation Node </div>' +
+        '<div> <i>' + relocating_node + '</i> </div>' +
+        '<div class="cr-custom-tooltip-label"> Relocation State </div>' +
+        '<div> <i>' + relocating_state + '</i> </div>' +
+        '<div class="cr-custom-tooltip-label"> State </div>' +
+        '<div> <i>' + state + '</i> </div>' +
+        '</div> ' +
+        '</div>';
+    }
+
+    $rootScope.$on('shard-query-done', function () {
+      $scope.response = ShardsIntervalService.response;
+      $scope.formatShardData();
+    });
+
+    $scope.formatShardData = function () {
+      $scope.tables = {};
+      $scope.tables_partitions = [];
+      $scope.schemas = {};
+      $scope.shards = {};
+
+      var formatted_response = {};
+
+      $scope.response.forEach(function (obj) {
+
+        if (!obj.node_id) {
+          obj.node_id = 'unassigned';
+        }
+
+        if (!formatted_response[obj.node_id]) {
+          formatted_response[obj.node_id] = {};
+        }
+
+        if (!formatted_response[obj.node_id].shards_by_table) {
+          formatted_response[obj.node_id].shards_by_table = {};
+        }
+
+        if (!formatted_response[obj.node_id].shards_by_table[obj.key]) {
+          formatted_response[obj.node_id].shards_by_table[obj.key] = {};
+        }
+
+        if (!formatted_response[obj.node_id].shards_by_table[obj.key].shards) {
+          formatted_response[obj.node_id].shards_by_table[obj.key].shards = [];
+        }
+
+        formatted_response[obj.node_id].shards_by_table[obj.key].shards.push({
+          'state_class': $filter('shardStateClass')(obj.state, obj.primary),
+          'html': getHtmlStringForShard(obj.id, obj.size, obj.min_lucene_version, obj.num_docs, obj.primary, obj.relocating_node, obj.routing_state, obj.state, $filter('shardStateClass')(obj.state, obj.primary))
```

Comment:
I'd rather use `angular.$compile` instead of plain string manipulation
there are also ways to load a html file as template. this would remove the html code from the JS which is desirable IMO :)